### PR TITLE
eth: add chaindbProperty to debug API (+ console fixes)

### PR DESCRIPTION
**Pull request by fjl**
I wanted to look at leveldb properties and added the API, but then I got carried
away and fixed a few things in the console, too.

chaindbproperty

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Sat Feb 20 14:03:05 UTC 2016_

File: cmd/utils/jeth.go [hunk 1]
```diff
@@ -129,76 +110,62 @@ func (self *Jeth) NewAccount(call otto.FunctionCall) (response otto.Value) {
...
-
-		response, err = call.Otto.Run(`
-		ret_response[response_idx] = { jsonrpc: ret_jsonrpc, id: ret_id, result: JSON.parse(ret_result) };
-		`)
+	req := json.RawMessage(reqv.String())
+
+	if cb := call.Argument(1); cb.IsFunction() {
+		// Asynchronous call: do the the communication on another goroutine, then switch
+		// back to the event loop and call cb with the response or error.
+		go self.execRequests(&req, func(resp json.RawMessage, err error) {
```
Comment (bas-vk) [inline cmd/utils/jeth.go]: The RPC layer is able to execute requests in parallel and can send responses back out of order. It is up to the caller to correlate a response with the requests and call the appropriate callback. This mechanism doesn't match the response id with the request id and can call the wrong callback for the response.

Example:
```
> eth.getAccounts(function(e, accs) { console.log(accs) }); eth.blockNumber
1042569
> 0x079bd9292ae3f362de496bb706994bb13f0f66be,0x66f183060253cfbe45beff1e6e7ebbe318c81e56,0x4ba70599fa368eb9b0720ff6af21315f630a13f1


> eth.getAccounts(function(e, accs) { console.log(accs) }); eth.blockNumber
0xfe889
new BigNumber() not a base 16 number: 
    at L (bignumber.js:3:2876)
    at bignumber.js:3:8435
    at a (bignumber.js:3:389)
    at bignumber.js:3:8425
    at a (bignumber.js:3:796)
    at a (bignumber.js:3:228)
    at web3.js:2236:16
    at web3.js:2097:12
    at web3.js:5701:54
    at get (web3.js:5751:16)
    at <anonymous>:1:1=
```

File: cmd/utils/jeth.go [hunk 2]
```diff
@@ -129,76 +110,62 @@ func (self *Jeth) NewAccount(call otto.FunctionCall) (response otto.Value) {
...
-		call.Otto.Set("ret_result", string(payload))
-		call.Otto.Set("response_idx", i)
-
-		response, err = call.Otto.Run(`
-		ret_response[response_idx] = { jsonrpc: ret_jsonrpc, id: ret_id, result: JSON.parse(ret_result) };
-		`)
+	req := json.RawMessage(reqv.String())
+
+	if cb := call.Argument(1); cb.IsFunction() {
+		// Asynchronous call: do the the communication on another goroutine, then switch
```
Comment (bas-vk) [inline cmd/utils/jeth.go]: `do the the communication` -> `do the communication`

File: cmd/utils/jeth.go (hunk 1 above)
Comment (fjl) [inline cmd/utils/jeth.go]: Damn. I didn't think about that. Will fix.

Comment (bas-vk) [inline cmd/utils/jeth.go]: If you work on a solution keep in mind that we also have subscriptions. I'm not sure how these will be implemented in web3 therefore I didn't change anything yet.

**Comment by bas-vk** [comment]
This PR will also fix https://github.com/ethereum/go-ethereum/issues/2230

**Comment by obscuren** [comment]
@fjl status?

**Comment by fjl** [comment]
I have removed the web3 provider rewrite (we'll do this when the new RPC client is done) and rebased on develop.

**Comment by fjl** [comment]
@bas-vk @obscuren please review

**Comment by fjl** [comment]
rebased on develop

**Comment by bas-vk** [comment]
:+1:

**Comment by obscuren** [comment]
:+1: LGTM. Can't find how this fixes #2230 though

**Comment by fjl** [comment]
It used to fix this issue because the console web3 provider changed, but I removed that. However bas has already solved it in another way.

**Comment by obscuren** [comment]
Would have been nice if you had squashed first

**Pull request merged**